tools/windows/DatadogAgentInstaller/CustomActions/RegistryCustomActions.cs

```diff
@@ -85,11 +86,26 @@ private static ActionResult ReadRegistryProperties(ISession session)
                                 return string.Empty;
                             });
 
-                        // PROJECTLOCATION
                         RegistryValueProperty(session, "PROJECTLOCATION", subkey, "InstallPath");
-
-                        // APPLICATIONDATADIRECTORY
                         RegistryValueProperty(session, "APPLICATIONDATADIRECTORY", subkey, "ConfigRoot");
+                        RegistryValueProperty(session, "ALLOWCLOSEDSOURCE", subkey, "AllowClosedSource");
+                    }
+                }
+
+                if (string.IsNullOrEmpty(session.Property("ALLOWCLOSEDSOURCE")))
+                {
+                    session.Log("Cannot find the \"AllowClosedSource\" registry key, checking the NPM service state.");
+                    using var subkey = Registry.LocalMachine.OpenSubKey(@"SYSTEM\CurrentControlSet\Services\ddnpm");
+                    if (subkey != null)
+                    {
+                        // https://renenyffenegger.ch/notes/Windows/registry/tree/HKEY_LOCAL_MACHINE/System/CurrentControlSet/Services/_driver_name_/index
+                        // Start = 4 means service is disabled, so NOT 4 means it's enabled.
+                        session["ALLOWCLOSEDSOURCE"] = int.Parse(subkey.GetValue("Start").ToString()) != 4 ? "1" : "0";
```

Comment:
Use [ServiceController.StartType](https://learn.microsoft.com/en-us/dotnet/api/system.serviceprocess.servicecontroller.starttype?view=dotnet-plat-ext-7.0) instead of checking the registry.